### PR TITLE
TaskSkill model, policies and endpoints

### LINKED ISSUE #670: Add TaskSkill model and endpoints
# Problem

To be able to match tasks with users, we need a `TaskSkill` model with associated endpoints, policies, etc.

## Model
* [x] `TaskSkill` model
  * `[:skill_id, :task_id]`
  * `create_changeset` - requires both fields. uniqueness on that specific combination of fields

## Endpoints
* [x] `POST /task-skills` to assign a skill to a project - creates record
* [x] `DELETE /task-skills` to remove a skill from a project - deletes record
* [x] `GET /task-skills` - to fetch all project skills (with coalesce support for `hasMany`)
* [x] `GET /task-skills/:id`

## Policies (Authorization)

* [x] CREATE and DELETE should be task author, project admin or higher. **Up for discussion**. Possibly just a `contributor` might be enough.
* [x] SHOW and INDEX should be public

## Authentication

* [x] CREATE and DELETE should require authentication
* [x] SHOW and INDEX should be public

Progress on: #668

### PR DESCRIPTION
**Pull request by begedin**
# What's in this PR?

Added required infrastructure to create, delete and view TaskSkill records

## References
Fixes #670
Progress on: #667

### CODE REVIEW

File: test/policies/task_skill_policy_test.exs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    end
+
+    test "returns false when user is pending member of organization" do
+      {user, task} = generate_data_for("pending")
+
+      changeset = %TaskSkill{} |> create_changeset(%{task_id: task.id})
+      refute create?(user, changeset)
+    end
+
+    test "returns false when user is contributor of organization" do
```
Comment (joshsmith) [inline test/policies/task_skill_policy_test.exs]: I think we want this to be `true` for contributors.

File: test/policies/task_skill_policy_test.exs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    test "returns false when user is pending member of organization" do
+      {user, task} = generate_data_for("pending")
+
+      task_skill = insert(:task_skill, task: task)
+
+      refute delete?(user, task_skill)
+    end
+
+    test "returns false when user is contributor of organization" do
+      {user, task} = generate_data_for("contributor")
```
Comment (joshsmith) [inline test/policies/task_skill_policy_test.exs]: `true` here, too.

**Review by joshsmith — CHANGES_REQUESTED** [review]
Minor change requested here.

**Comment by begedin** [comment]
@JoshSmith Made the change, rebased, should be good to go.

**Review by joshsmith — APPROVED** [review]

**Pull request merged**